File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
        
    
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        
        def dfs(node1,node2):
            if not node1:
                return None 
            
            elif target.val ==  node1.val :
                return node2
            
            return dfs(node1.left,node2.left) or dfs(node1.right,node2.right ) 
        
        return dfs(original,cloned)

File: 1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1379-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    
    TreeNode* dfs(TreeNode* clone,TreeNode* target){
        
        if(!clone)
            return nullptr;
        
        if(clone->val == target->val)
            return clone;
        
        TreeNode* left = dfs(clone->left, target);
        TreeNode* right = dfs(clone->right, target) ;
        
        
        return left!=nullptr? left : right;
    
    }
    
    
    TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {
        
        return dfs(cloned,target);    
        
    
    
    }
};